### ISSUE TITLE
[Azure Module] Analyze memory consumption peak and post peak (standby) - Linux

### ISSUE THREAD

**Issue by tinachoB** [wazuh/wazuh#9303]
## Description
This issue aims to research and find out how much memory the component might take in a normal product use.

## DoD
- [ ] Test, track and analyze result while azure module is running

**Comment by tinachoB** [comment]
Using Massif tool and stimulating the azure module with CI/CD configuration parameters we got the following diagrams (depending the different processes being involved):

**azure-logs** 
![image](https://user-images.githubusercontent.com/22640902/126021436-2893ed5e-3402-437b-858c-0f371d0030b5.png)
![image](https://user-images.githubusercontent.com/22640902/126021445-87ca083c-edb9-4b5d-8b95-25b099695ec3.png)

**wazuh-modulesd**
![image](https://user-images.githubusercontent.com/22640902/126021506-2844e6ba-8796-4a32-9009-bee4cc9aefdf.png)
![image](https://user-images.githubusercontent.com/22640902/126021513-c5642397-cfd4-4b4f-9ab5-c7bb2b143ec5.png)

**Comment by tinachoB** [comment]
**Final analysis:**

The following results were obtained **with all Wazuh modules disabled except by `azure-logs` wodle.**

![image](https://user-images.githubusercontent.com/22640902/126400094-3ff3a954-1d09-4085-bae2-7d1d7e2acf2d.png)

In the the above graph the peak memory usage is produced when the `libwazuhext.so` is being called for yaml related operations:
![image](https://user-images.githubusercontent.com/22640902/126400355-c21a8cab-5809-4251-b568-ea580835cfb4.png).

Regarding with azure module itself, the following call stack has been observed in different tests:
![image](https://user-images.githubusercontent.com/22640902/126400542-a6ca103d-69d2-49f9-8587-01403575feb2.png)

This is related to the configuration reading in order to properly execute the `azure-logs` wodle.

Once the configuration is being loaded, the `azure-logs` execution (for detailed information about the step by step of this refer to the [azure sequence diagram](https://github.com/wazuh/wazuh/issues/9301#issuecomment-881641957)) consumes the following memory:

![image](https://user-images.githubusercontent.com/22640902/126400991-1a500d94-7f19-4239-b707-a0a9add383fd.png)

The peak here is related to the azure-logs binary execution and the `libc` and `bash` interaction primarely:
![image](https://user-images.githubusercontent.com/22640902/126401241-52a3d07b-0ce3-4232-a311-d20cae51c781.png)

Both graphs show a very low memory consumption while the module is being executed. 

To sum up, there is no memory issues considering azure module over these tests.
